File: src/Easy/Leetcode121.java
package Easy;

public class Leetcode121 {
    public int maxProfit(int[] prices) {
        int minStockPrice=prices[0];
        int maxProfit=0;
        for(int price:prices){
            maxProfit=Math.max(maxProfit,price-minStockPrice);
            minStockPrice=Math.min(minStockPrice,price);
        }
        return maxProfit;
    }
}

File: src/Python/Easy/Leetcode121.py
class Solution:
    def maxProfit(self, prices: List[int]) -> int:
        minPrice=prices[0]
        maxProfit=0

        for num in prices:
            maxProfit=max(maxProfit,num-minPrice)
            minPrice=min(minPrice,num)

        return maxProfit
